anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/source/data/metrictank/MetricTankClient.java

```diff
@@ -53,35 +55,43 @@
     /**
      * Fetch metric data for a given set of metrics
      *
-     * @param from          earliest time
-     * @param target        metric name or tag with an optional graphite function
+     * @param from   earliest time
+     * @param target metric name or tag with an optional graphite function
      * @return time series for the specified metric
      */
-    public List<GraphiteResult> getData(long from, long until, String target) {
+    public List<MetricTankResult> getData(long from, long until, String target) {
 
         notNull(from, "from can't be null");
         notNull(until, "until can't be null");
         notNull(target, "target can't be null");
 
         val uri = String.format(baseUri + FETCH_METRICS_PATH, from, until, target);
         log.debug("Sending query to Graphite target: {}", uri);
+        
         Content content;
+        val headers = buildHeaders();
         try {
-            content = httpClient.get(uri);
+            content = httpClient.get(uri, headers);
         } catch (IOException e) {
             val message = String.format("Encountered IOException while querying Graphite target '%s': httpMethod=GET, uri=%s, message=%s",
                     target,
                     uri,
                     e.getMessage());
-            throw new GraphiteClientException(message, e);
+            throw new MetricTankClientException(message, e);
         }
 
-        List<GraphiteResult> results;
+        List<MetricTankResult> results;
         try {
-            results = Arrays.asList(objectMapper.readValue(content.asBytes(), GraphiteResult[].class));
+            results = Arrays.asList(objectMapper.readValue(content.asBytes(), MetricTankResult[].class));
         } catch (IOException e) {
-            throw new GraphiteClientException(String.format("IOException while parsing response from Graphite target: %s", target), e);
+            throw new MetricTankClientException(String.format("IOException while parsing response from Graphite target: %s", target), e);
         }
         return results;
     }
+
+    private Map<String, String> buildHeaders() {
+        val headers = new HashMap<String, String>();
+        headers.put("x-org-id", "1");
+        return headers;
```

Comment:
Does this map need to be mutable? If not, prefer singletonMap:
https://stackoverflow.com/questions/7125536/when-would-i-use-java-collections-singletonmap-method

Reply:
good call. updating it to singletonMap